src/registration/metric/demons.h

```diff
@@ -47,6 +51,7 @@ namespace MR
           }
 
           ~Demons () {
+            std::lock_guard<std::mutex> lock (*mutex);
```

Comment:
Pretty sure that only locks decrementing the thread counter and erasing the `__Backend` class (returning to non-thread-safe `stderr` printing) if it's the last thread that's being unregistered; doesn't provide any guarantee on class destructors. I recall having fixed at least one bug in multi-threading apps that arose because of the absence of mutex locking in the destructor when thread-wise stats were being summated. If safety against such race conditions is now provided in the backend, either implicitly or explicitly, it's not where you've highlighted.